packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -116,19 +117,24 @@ function useBlockEditorSettings( settings, postType, postId ) {
 				getEntityRecord,
 				getUserPatternCategories,
 				getBlockPatternCategories,
+				getCurrentTheme,
 			} = select( coreStore );
 			const { get } = select( preferencesStore );
 			const { getBlockTypes } = select( blocksStore );
 			const siteSettings = canUser( 'read', 'settings' )
 				? getEntityRecord( 'root', 'site' )
 				: undefined;
-
+			const _currentTheme = getCurrentTheme();
 			return {
 				allowRightClickOverrides: get(
 					'core',
 					'allowRightClickOverrides'
 				),
 				blockTypes: getBlockTypes(),
+				currentTheme: {
```

Comment:
Here's where my lack of imagination or sleep failed me. I couldn't think of another way, though there probably is one, to ensure the current theme data is available in the editor context.